### PR TITLE
ICMSLST-2063 - Import and refactor of import files to s3 code

### PR DESCRIPTION
**Pull request by marcuspp**
Command to import files into s3
 - Adds command
 - Adds tests for command
 - Supported options `--limit` `--ignore-last-run` `--queries` `--s3-file-prefix` `--number-of-rows` `--count-only` 

Known issues (other tasks still to do)
 - Command is using the wrong queries and should use the queries found in `files.py`
 - The last run data is only written to s3 on a successful run and not if an error occurs

### CODE REVIEW

File: data_migration/management/commands/import_v1_files_to_s3.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            "--count-only",
+            help="Retrieves a count of number of files that will be processed per query",
+            action="store_true",
+            default=False,
+        )
+
+    def get_db(self, options) -> OracleDBProcessor:
+        """
+        Returns a DB processor that facilitates executing database queries on the v1 oracle database.
+        """
```
Comment (MattHolmes123) [inline data_migration/management/commands/import_v1_files_to_s3.py:69]: Can we do pep8 style docstrings please (same applices to all your docstrings):
https://peps.python.org/pep-0008/#documentation-strings

File: data_migration/management/commands/utils/db_processor.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from data_migration.queries.files_v1 import QUERY_LIST
+
+
+class OracleDBProcessor:
+    QUERIES: list[dict] = QUERY_LIST
+
+    def __init__(self, limit, selected_queries, number_of_rows):
+        self.limit = limit
+        self.selected_queries = selected_queries
+        self.number_of_rows = number_of_rows
```
Comment (MattHolmes123) [inline data_migration/management/commands/utils/db_processor.py:16]: Missing types

File: data_migration/management/commands/utils/db_processor.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    def get_sql(self, query_dict: dict[str, str], start_from_datetime: str) -> str:
+        sql = self.get_basic_sql(query_dict, start_from_datetime)
+        sql = self.add_limit_to_sql(sql)
+        return self.add_select_to_sql(sql)
+
+    def get_db_connection(self) -> oracledb.Connection:
+        return oracledb.connect(**CONNECTION_CONFIG)
+
+    @contextmanager
+    def execute_query(self, sql: str) -> Iterator:
```
Comment (MattHolmes123) [inline data_migration/management/commands/utils/db_processor.py:50]: We know the column names that the queries all return so we could use a TypedDict if you wanted extra type checking where it's used but it's not that important.

File: data_migration/management/commands/import_v1_files_to_s3.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        Uploads a file to s3 adding a prefix to the file name (key) if required.
+        """
+        if self.file_prefix:
+            _key = f"{self.file_prefix}/{_key}"
+        s3_web.upload_file_obj_to_s3(_file_obj, _key)
+
+    def get_initial_run_data_dict(
+        self, query_name, number_of_files_to_be_processed, start_from_datetime
+    ) -> dict:
```
Comment (MattHolmes123) [inline data_migration/management/commands/import_v1_files_to_s3.py:87]: Missing types.

File: data_migration/queries/files_v1.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        "query": QUERY_5,
+        "query_name": "V1_QUERY_5",
+    },
+    {
+        "query": QUERY_6,
+        "query_name": "V1_QUERY_6",
+    },
+]
+
+AVAILABLE_QUERIES = [query["query_name"] for query in QUERY_LIST]
```
Comment (MattHolmes123) [inline data_migration/queries/files_v1.py:224]: Where `QUERY_LIST` and `AVAILABLE_QUERIES` are used couldn't you do it all with a dictionary?
e.g.
```
QUERIES = {
    "V1_QUERY_1": QUERY_1,
    "V1_QUERY_2": QUERY_2,
    "V1_QUERY_3": QUERY_3,
    "V1_QUERY_4": QUERY_4,
    "V1_QUERY_5": QUERY_5,
    "V1_QUERY_6": QUERY_6,
}    

AVAILABLE_QUERIES = QUERIES.keys()
```

and when iterating over them:
```
    def get_query_list(self) -> Iterator[dict]:
        for name, query in QUERIES.items():
            if not self.selected_queries or (name in self.selected_queries):
                yield query
```

**Review by MattHolmes123 — APPROVED** [review]

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**